www/js/services/shareLocalDBFile.ts

```diff
@@ -0,0 +1,116 @@
+          // logDebug(`fileEntry ${fileEntry.nativeURL} is file? ${fileEntry.isFile.toString()}`);
+          logDebug(`fileEntry is: ${JSON.stringify(fileEntry, null, 2)}`);
+          window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (copyDir) => {
+            logDebug(`DirectoryEntry is: ${JSON.stringify(copyDir.filesystem.root, null, 2)}`);
+
+            fileEntry.copyTo(
+              copyDir.filesystem.root,
+              fileName + fileExtension,
+              (res) => {
+                logDebug(`Res: ${res}`);
+                resolve();
+              },
+              (rej) => {
+                logDebug(`Rej: ${JSON.stringify(rej, null, 2)}`);
+                reject();
+              },
+            );
+          });
+        });
+      });
+    });
+  }
+
+  function localShareFile() {
+    return new Promise<void>((resolve, reject) => {
+      window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (fs) => {
+        fs.filesystem.root.getFile(
+          fileName + fileExtension,
+          null,
+          (fileEntry) => {
+            const shareObj = {
+              files: [fileEntry.nativeURL],
+              message: i18next.t('shareFile-service.send-log.body-please-fill-in-what-is-wrong'),
+              subject: i18next.t('shareFile-service.send-log.subject-logs'),
+            };
+            window['plugins'].socialsharing.shareWithOptions(
+              shareObj,
+              (result) => {
+                logDebug(`Share Completed? ${result.completed}`); // On Android, most likely returns false
+                logDebug(`Shared to app:  ${result.app}`);
+                resolve();
+              },
+              (msg) => {
+                logDebug(`Sharing failed with message ${msg}`);
+              },
+            );
+          },
+          (error) => {
+            displayError(error, 'Error while sharing logs');
+            reject(error);
+          },
+        );
+      });
+    });
+  }
+
+  function localClearData() {
+    return new Promise<void>((resolve, reject) => {
+      window['resolveLocalFileSystemURL'](window['cordova'].file.cacheDirectory, (fs) => {
+        fs.filesystem.root.getFile(fileName + fileExtension, null, (fileEntry) => {
+          fileEntry.remove(
+            () => {
+              logDebug(`Successfully cleaned up file ${fileName}`);
+              resolve();
+            },
+            (err) => {
+              logWarn(`Error deleting ${fileName} : ${err}`);
+              reject(err);
+            },
+          );
+        });
+      });
+    });
+  }
+
+  return {
+    copyFile: localCopyFile,
+    shareData: localShareFile,
+    clearData: localClearData,
+  };
+}
+export async function sendLocalDBFile(database: string) {
+  alert(i18next.t('shareFile-service.send-to'));
+
+  const dataMethods = localDBHelpers(database);
+  dataMethods
+    .copyFile()
+    .then(dataMethods.shareData)
+    .then(dataMethods.clearData)
```

Comment:
why are we clearing the data here after sharing? That is not what we do now. If we clear the data after sharing, then sharing the userCacheDB will result in collected and locally cached data getting deleted. Similarly, logs will be deleted after sharing and will not wait until they age out

Reply:
TL;DR: because we need to copy the file to add the `.txt` extension, we need to delete the duplicate file.

---

Full Explanation: Because we can't rename the file, we instead need to copy it (adding the `.txt`), and share that new file.  Per the plugin documentation ([link](https://github.com/apache/cordova-plugin-file/blob/03a4f4cb53e54de31344851cc5d01b5da96aa0d7/types/index.d.ts#L100))...
>  A copy of a file on top of an existing file must attempt to delete and replace that file.

This was an issue I ran into during development (see [this](https://github.com/e-mission/e-mission-phone/pull/1160#issuecomment-2192454813) comment) - if we do not clear space for this temporary file, the copy function will fail.  

> Similarly, logs will be deleted after sharing and will not wait until they age out

If I understand your concern correctly, this shouldn't be an issue.  We never delete the "main" logs -- we create a copy at the time of sharing, and then clear that copy.  (FWIW, Louis tested sending the logs twice in a row, and we had no issue with old logs being cleared before aging out).  

Perhaps the `clearData` function could be renamed to better explain its purpose, `clearTempFile`, etc.

Reply:
Yes, please rename to clarify